File: java/eclipse/Travel/numberofUniquePaths/NumberofUniquePaths.java
package numberofUniquePaths;

import java.util.Scanner;

public class NumberofUniquePaths  {
	static Scanner scanner = new Scanner(System.in);

	public static void main(String[] args) {
		int test = scanner.nextInt();
		for (int i = 0; i < test; ++i)
			test_case();
	}

	static void test_case() {
		int m = scanner.nextInt();
		int n = scanner.nextInt();

		System.out.println(function(m, n));
	}

	private static int function(int m, int n) {

		int arr[][] = new int[m + 1][n + 1];
		for (int i = 1; i <= m; ++i) {
			for (int j = 1; j <= n; ++j) {
				if (i == 1 || j == 1)
					arr[i][j] = 1;
				else
					arr[i][j] = arr[i - 1][j] + arr[i][j - 1];
			}
		}
		return arr[m][n];
	}
}

File: cpp/Dynamic Programming/number of unique paths.cpp
#include <iostream>

int recu(int m, int n) {
    if (m == 1 || n == 1)
        return 1;
    return recu(m - 1, n) + recu(m, n - 1);
}

void test_case() {
    int m, n;
    std::cin >> m >> n;
    std::cout << recu(m, n) << std::endl;
}

int main() {
    int test;
    std::cin >> test;
    for (int i = 0; i < test; ++i)
        test_case();
    return 0;
}
